packages/esm-ward-app/src/ward-workspace/ward-patient-notes/form/notes-form.component.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import dayjs from 'dayjs';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Controller, useForm } from 'react-hook-form';
+import { Button, Column, Form, InlineLoading, InlineNotification, Row, Stack, TextArea } from '@carbon/react';
+import {
+  createErrorHandler,
+  type DefaultWorkspaceProps,
+  type PatientUuid,
+  ResponsiveWrapper,
+  showSnackbar,
+  translateFrom,
+  useSession,
+} from '@openmrs/esm-framework';
+import { savePatientNote } from './notes-form.resource';
+import styles from './notes-form.scss';
+import { moduleName } from '../../../constant';
+import useEmrConfiguration from '../../../hooks/useEmrConfiguration';
+
+type NotesFormData = z.infer<typeof noteFormSchema>;
+
+const noteFormSchema = z.object({
+  wardClinicalNote: z.string().refine((val) => val.trim().length > 0, {
+    //t('clinicalNoteErrorMessage','Clinical note is required')
+    message: translateFrom(moduleName, 'clinicalNoteErrorMessage', 'Clinical note is required'),
+  }),
+});
+
+interface PatientNotesFormProps extends DefaultWorkspaceProps {
+  patientUuid: PatientUuid;
+}
+
+const PatientNotesForm: React.FC<PatientNotesFormProps> = ({
+  closeWorkspaceWithSavedChanges,
+  patientUuid,
+  promptBeforeClosing,
+}) => {
+  const { emrConfiguration, isLoadingEmrConfiguration, errorFetchingEmrConfiguration } = useEmrConfiguration();
+  const { t } = useTranslation();
+  const session = useSession();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [rows, setRows] = useState<number>();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isDirty },
+  } = useForm<NotesFormData>({
+    mode: 'onSubmit',
+    resolver: zodResolver(noteFormSchema),
+    defaultValues: {
+      wardClinicalNote: '',
+    },
+  });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty, promptBeforeClosing]);
+
+  const locationUuid = session?.sessionLocation?.uuid;
+  const providerUuid = session?.currentProvider?.uuid;
+
+  const onSubmit = useCallback(
+    (data: NotesFormData) => {
+      const { wardClinicalNote } = data;
+      setIsSubmitting(true);
+
+      const notePayload = {
+        encounterDatetime: dayjs(new Date()).format(),
```

Comment:
Don't send encounterDatetime from the server. It is automatically added by the BE server. This will otherwise put us into issues like this: https://github.com/openmrs/openmrs-ngx-formentry/pull/131